autoload/ale/handlers/haskell.vim

```diff
@@ -1,5 +1,15 @@
 " Author: w0rp <devw0rp@gmail.com>
 " Description: Error handling for the format GHC outputs.
+"
+function! ale#handlers#haskell#GetStackExecutable(bufnr) abort
+    if ale#path#FindNearestFile(a:bufnr, 'stack.yaml') isnot# ''
+        return 'stack'
+    else
+        " if there is no stack.yaml file, we don't use stack even if it exists,
+        " so we return '', because executable('') apparently always fails
+        return ''
+    endif
```

Comment:
Move `return ''` to the end of the function, instead of putting it inside `else`.

Reply:
Fixed